infra/base-images/base-runner/bad_build_check

```diff
@@ -15,44 +15,44 @@
 #
 ################################################################################
 
-# Verify that the given fuzzer has proper coverage instrumentation.
+# Global variable that can be overwritten by any check.
+BAD_BUILD_CHECK_FAILED=0
+
+# Verify that the given fuzz target has proper coverage instrumentation.
 function check_instrumentation {
   local FUZZER=$1
+  local LOG_PATH_FOR_BROKEN_TARGET=$2
   local CHECK_FAILED=0
+  local FUZZER_OUTPUT="/tmp/$(basename $FUZZER).output"
 
   if [[ "$FUZZING_ENGINE" = libfuzzer ]]; then
-    CHECK_FAILED=$($FUZZER -max_total_time=4 2>&1 | egrep "ERROR: no interesting inputs were found. Is the code instrumented" -c)
+    $FUZZER -runs=4 &>$FUZZER_OUTPUT
+    CHECK_FAILED=$(cat $FUZZER_OUTPUT | egrep "ERROR: no interesting inputs were found. Is the code instrumented" -c)
     if (( $CHECK_FAILED > 0 )); then
-      echo "BAD BUILD: the code does not seem to have coverage instrumentation."
-    fi
-  fi
+      echo "BAD BUILD: the target does not seem to have coverage instrumentation." >> $LOG_PATH_FOR_BROKEN_TARGET
+      BAD_BUILD_CHECK_FAILED=1
 
-  # honggfuzz is not fully integrated, avoid performing the following check.
-  if [[ "$FUZZING_ENGINE" != honggfuzz ]]; then
-    if (( $CHECK_FAILED == 0 )); then
-      # The "example" target has 93 when built with ASan and 67 with UBSan. Real
-      # targets have greater values (arduinojson: 413, libteken: 519, zlib: 586).
-      local THRESHOLD_FOR_NUMBER_OF_EDGES=90
-      if [[ $SANITIZER = undefined ]]; then
-        THRESHOLD_FOR_NUMBER_OF_EDGES=65
-      fi
-
-      local NUMBER_OF_EDGES=$(sancov -print-coverage-pcs $FUZZER | wc -l)
-      if (( $NUMBER_OF_EDGES < $THRESHOLD_FOR_NUMBER_OF_EDGES )); then
-        echo "BAD BUILD: the code does not seem to have coverage instrumentation."
-        CHECK_FAILED=1
-      fi
+      # Bail out as the further check does not make any sense, there are 0 PCs.
+      return 1
     fi
-  fi
 
-  if (( $CHECK_FAILED > 0 )); then
-    exit 1
+    local NUMBER_OF_EDGES=$(cat $FUZZER_OUTPUT | grep -Po "INFO: Loaded [[:digit:]]+ module.*\(.*counters\):[[:space:]]+\K[[:digit:]]+")
+
+    # The "example" target has 73 with ASan, 65 with UBSan, and 6648 with MSan.
+    # Real world targets have greater values (arduinojson: 407, zlib: 664).
+    local THRESHOLD_FOR_NUMBER_OF_EDGES=256
+
+    if (( $NUMBER_OF_EDGES < $THRESHOLD_FOR_NUMBER_OF_EDGES )); then
+      echo "BAD BUILD: the target seems to have only partial coverage instrumentation." >> $LOG_PATH_FOR_BROKEN_TARGET
+      BAD_BUILD_CHECK_FAILED=1
+    fi
   fi
 }
 
-# Verify that the given fuzzer has been built properly and works as expected.
+# Verify that the given fuzz target has been built properly and works.
 function check_startup_crash {
   local FUZZER=$1
+  local LOG_PATH_FOR_BROKEN_TARGET=$2
   local CHECK_PASSED=0
 
   if [[ "$FUZZING_ENGINE" = libfuzzer ]]; then
```

Comment:
Lets not use hardcoded 4 runs inside this if.

Reply:
Done.